packages/flutter/lib/src/painting/debug.dart

```diff
@@ -48,19 +47,19 @@ class ImageSizeInfo {
 
   /// A unique identifier for this image, for example its asset path or network
   /// URL.
-  final String source;
+  final String? source;
 
   /// The size of the area the image will be rendered in.
-  final Size displaySize;
+  final Size? displaySize;
 
   /// The size the image has been decoded to.
-  final Size imageSize;
+  final Size? imageSize;
```

Comment:
I looked at the usage of `ImageSizeInfo` and was surprised that we allow `null` for these values.

Reply:
I think this should not be null

Reply:
So I have to make a "breaking change" with:

```diff
-  const ImageSizeInfo({this.source, this.displaySize, this.imageSize});
+  const ImageSizeInfo({this.source, this.displaySize, required this.imageSize});
```

Is it ok for you?

Reply:
This type was very recently added and only used for debugging purposes.  It should be fine.

Reply:
Actually there are several tests failures with this change and I'm not confortable how to change them. Could you made this change after this PR is merged?

Reply:
That's fine

Reply:
Sorry I made a mistake. The failures appear with `ImageInfo` where some tests return null image

Reply:
The tests fail with the following diff:

```diff
--- a/packages/flutter/lib/src/painting/image_cache.dart
+++ b/packages/flutter/lib/src/painting/image_cache.dart
@@ -389,7 +389,7 @@ class ImageCache {
     _PendingImage? untrackedPendingImage;
     void listener(ImageInfo? info, bool syncCall) {
       // Images that fail to load don't contribute to cache size.
-      final int imageSize = info == null || info.image == null ? 0 : info.image.height * info.image.width * 4;
+      final int imageSize = info == null ? 0 : info.image.height * info.image.width * 4;
 
       final _CachedImage image = _CachedImage(result!, imageSize);
```